apps/openmw/mwmechanics/character.cpp

```diff
@@ -382,8 +382,9 @@ void CharacterController::refreshJumpAnims(const WeaponInfo* weap, JumpingState
         }
         else
         {
-            mAnimation->disable(mCurrentJump);
-            mCurrentJump.clear();
+            if (startAtLoop) 
```

Comment:
I don't quite understand why the `if (startAtLoop)` was added.

Also, aren't we missing an `mCurrentJump = jumpAnimName` here?

Reply:
`mCurrentJump = jumpAnimName` was never (it was cleared instead) there so I just kept it that way as I only wanted to fix the issue at hand. Actually the clearing of `mCurrentJump` had its purpose (see below).

`if (startAtLoop)` is there to only disable the current animation at the frame the landing animation is to be started. Previously this was achieved by clearing `mCurrentJump` (so that the next updates call disable on empty string animation) but as I prevent it from being cleared now, I have to test it this way. Without this test the landing animation gets stopped in the next update.

Reply:
Hm, the current code is clearly a little odd. Seemingly the JumpState_Landing only gets set for one frame and then reset to JumpState_None despite the landing animation still playing. And then we call play() with an empty animation name. 

May I suggest try to refactor the code so the states are set properly and mCurrentJump always refers to currently played animation?

Reply:
>  Seemingly the JumpState_Landing only gets set for one frame

Yes it does. Okay, I'll try to refactor this, at least I'll get more understanding of the animation system.